stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -0,0 +1,634 @@
+module Artifacts
+
+import Base: get, SHA1
+using Base.BinaryPlatforms, Base.TOML
+
+export artifact_exists, artifact_path, remove_artifact, verify_artifact,
+       artifact_meta, artifact_hash, find_artifacts_toml, @artifact_str
+
+"""
+    parse_toml(path::String)
+
+Uses Base.TOML to parse a TOML file
+"""
+function parse_toml(path::String)
+    p = Base.TOML.Parser()
+    Base.TOML.reinit!(p, read(path, String); filepath=path)
+    return Base.TOML.parse(p)
+end
+
+# keep in sync with Base.project_names and Base.manifest_names
+const artifact_names = ("JuliaArtifacts.toml", "Artifacts.toml")
+
+const ARTIFACTS_DIR_OVERRIDE = Ref{Union{String,Nothing}}(nothing)
+"""
+    with_artifacts_directory(f::Function, artifacts_dir::String)
+
+Helper function to allow temporarily changing the artifact installation and search
+directory.  When this is set, no other directory will be searched for artifacts, and new
+artifacts will be installed within this directory.  Similarly, removing an artifact will
+only effect the given artifact directory.  To layer artifact installation locations, use
+the typical Julia depot path mechanism.
+"""
+function with_artifacts_directory(f::Function, artifacts_dir::String)
+    try
+        ARTIFACTS_DIR_OVERRIDE[] = artifacts_dir
+        f()
+    finally
+        ARTIFACTS_DIR_OVERRIDE[] = nothing
+    end
+end
+
+"""
+    artifacts_dirs(args...)
+
+Return a list of paths joined into all possible artifacts directories, as dictated by the
+current set of depot paths and the current artifact directory override via the method
+`with_artifacts_dir()`.
+"""
+function artifacts_dirs(args...)
```

Comment:
Looks like this is only called with a `String` so far?

Reply:
Within this file, yes.  I could probably annotate `args` as `::AbstractString...`.

Reply:
So `args...` is really a path here? I don't get why it is accepting a vararg though, is it to not have the caller have to do a `joinpath`? Seems easier to read if it was just one argument and also likely better from the compiler point of view that you don't compile different versions for different number of arguments.

Reply:
Yeah, it was meant to mimic `abspath()` in that regard.  It does appear that I never ended up using that functionality, so I'll just restrict it to a single string.

Reply:
Aha, I figured out why I had this; it was actually as a convenient definition for `artifacts_dirs()`, which I use in `Pkg`.